deepchem/models/torch_models/layers.py

```diff
@@ -4792,3 +4792,276 @@ def forward(self, dist: torch.Tensor) -> torch.Tensor:
         dist = dist.unsqueeze(-1) / self.cutoff
         output: torch.Tensor = self.envelope(dist) * (self.freq * dist).sin()
         return output
+
+
+class MXMNetLocalMessagePassing(nn.Module):
+    """
+    The MXMNetLocalMessagePassing class defines a local message passing layer used in the MXMNet model [1]_.
+
+    This layer integrates cross-layer mappings inside the local message passing, allowing for the transformation
+    of input tensors representing pairwise distances and angles between atoms in a molecular system.
+    The layer aggregates information using message passing and updates atom representations accordingly.
+
+    The 3-step message passing scheme is proposed in the paper [1]_.
+
+    1. Step 1 contains Message Passing 1 that captures the two-hop angles and related pairwise distances to update edge-level embeddings {mji}.
+    2. Step 2 contains Message Passing 2 that captures the one-hop angles and related pairwise distances to further update {mji}. 
+    3. Step 3 finally aggregates {mji} to update the node-level embedding hi.
+    
+    These steps in the t-th iteration can be formulated as follows:
+
+    Let:
+        - **mlp** : ``MultilayerPerceptron``
+        - **res** : ``ResidualBlock``
+        - **h** : ``node_features``
+        - **m** : ``message with radial basis function``
+        - **idx_kj**: ``Tensor containing indices for the k and j atoms``
+        - **x_i** : ``The node to be updated``
+        - **h_i** : ``The hidden state of x_i``
+        - **x_j** : ``The neighbour node connected to x_i by edge e_ij``
+        - **h_j** : ``The hidden state of x_j``
+        - **rbf** : ``Input tensor representing radial basis functions``
+        - **sbf** : ``Input tensor representing the spherical basis functions``
+        - **idx_jj** : ``Tensor containing indices for the j and j'``
```

Comment:
Please check the description for "idx_jj"

Reply:
explain what is `j'`. Seems to me it represents atoms that are direct neighbours of `j` other than `i`. Please check.

Reply:
j' represents other neighbour nodes of i to get the other local angles.